persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/SoftIndexFileStore.java

```diff
@@ -163,40 +164,93 @@ public void start() {
       compactor.setIndex(index);
       startIndex();
       final AtomicLong maxSeqId = new AtomicLong(0);
-      if (index.isLoaded()) {
-         log.debug("Not building the index - loaded from persisted state");
-      } else if (configuration.purgeOnStartup()) {
-         log.debug("Not building the index - purge will be executed");
+      boolean migrateData = false;
+
+      if (!configuration.purgeOnStartup()) {
+         // we don't destroy the data on startup
+         // get the old files
+         FileProvider oldFileProvider = new FileProvider(getDataLocation(), configuration.openFilesLimit(), "");
+         if (oldFileProvider.hasFiles()) {
+            log.debug("Migrate old format to new format.");
```

Comment:
I think this log and a message indicating that the migration has sucessfully completed should be `info` the same as in SFS.

Reply:
should the log message be in the class logger or under `PERSISTENCE` logger?

Reply:
Good question. I think PERSISTENCE. The same applies for SFS